app/pods/newsletter/print/route.js

```diff
@@ -1,96 +1,75 @@
       sort: 'priority',
       'page[size]': 300,
     });
-    const announcements = this.filterAnnouncements(agendaitems.filter((agendaitem) => agendaitem.showAsRemark), params);
+    let notas = agendaitems.filter((agendaitem) => !agendaitem.showAsRemark);
+    let announcements = agendaitems.filter((agendaitem) => agendaitem.showAsRemark);
 
-    const {
-      draftAgendaitems, groupedAgendaitems,
-    } = await this.parseAgendaitems(
-      agendaitems, params
-    );
+    if (params.showDraft) {
+      notas = notas.sortBy('priority');
+      announcements = announcements.sortBy('priority');
+    } else { // Items need to be ordered by minister protocol order
+      const filteredNotas = await this.filterAgendaitems(notas);
 
-    await this.agendaService.groupAgendaitemsOnGroupName(draftAgendaitems);
+      // TODO: Below is a hacky way of grouping agendaitems for protocol order. Refactor.
+      await setCalculatedGroupPriorities(notas);
+      const groupedAgendaitems = Object.values(groupAgendaitemsByGroupname(notas));
+      await this.agendaService.groupAgendaitemsOnGroupName(filteredNotas);
 
-    const groupsArray = sortByPriority(groupedAgendaitems, this.allowEmptyGroups);
+      const itemGroups = sortByPriority(groupedAgendaitems, true); // An array of groups
+      notas = itemGroups.reduce((agendaItems, group) => agendaItems.push(group.agendaitems));
+    }
 
     return hash({
-      currentAgenda: agenda,
-      groups: groupsArray,
-      agendaitems: draftAgendaitems.sortBy('priority'),
-      announcements: announcements.sortBy('priority'),
-      meeting: session,
+      notas,
+      announcements,
     });
-  },
-
-  async parseAgendaitems(agendaitems, params) {
-    let draftAgendaitems = agendaitems.filter((agendaitem) => !agendaitem.showAsRemark && !agendaitem.isApproval);
-
-    draftAgendaitems = await this.filterAgendaitems(draftAgendaitems, params);
+  }
 
-    await setCalculatedGroupPriorities(draftAgendaitems);
+  setupController(controller) {
+    super.setupController(...arguments);
+    controller.set('meeting', this.modelFor('newsletter').meeting);
+  }
 
-    const groupedAgendaitems = Object.values(groupAgendaitemsByGroupname(draftAgendaitems));
-    return {
-      draftAgendaitems,
-      groupedAgendaitems,
-    };
-  },
-
-  filterAnnouncements(announcements) {
-    return announcements.filter((agendaitem) => agendaitem.showInNewsletter);
-  },
-
-  async filterAgendaitems(agendaitems, params) {
-    if (params.definite !== 'true') {
-      return agendaitems;
-    }
-    const newsLetterByIndex = await Promise.all(agendaitems.map(async(agendaitem) => {
+  async filterAgendaitems(agendaitems) {
+    const filteredAgendaitems = [];
+    for (const agendaItem of agendaitems) {
       try {
-        const agendaItemTreatment = await agendaitem.get('treatments').firstObject;
+        const agendaItemTreatment = await agendaItem.get('treatments').firstObject;
```

Comment:
geen gebruik maken van agendaItem met hoofdletter I plz.
agendaItemTreatment is de enige uitzondering hierop, gebaseerd op het model (agenda-item-treatment)
We willen geen nieuwe occurences van deze.

Reply:
M'n "ingebouwde camel-caser" at work :brain: :robot: ... heb aangepast.